.github/workflows/mobile-build.yml

```diff
@@ -0,0 +1,101 @@
+# Builds unity project and creates iOS and Android executables for sideloading.
+name: Mobile Build
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - '*'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      clean:
+        description: 'Clean the project before building'
+        required: false
+        default: false
+        type: 'boolean'
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ ( github.event_name == 'pull_request' || github.event.action == 'synchronize' ) }}
+jobs:
+  build:
+    env:
+      UNITY_PROJECT_PATH: '' # set by unity-setup action
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [windows-latest, macos-13]
+        include:
+          - os: windows-latest
+            build-target: Android
+            build-args: ''
+          - os: macos-13
+            build-target: iOS
+            build-args: ' -enableAppleAutomaticSigning'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          clean: ${{ github.event.inputs.clean == 'true' }}
+      - uses: buildalon/unity-setup@v1
+        with:
+          build-targets: ${{ matrix.build-target }}
+      - uses: buildalon/activate-unity-license@v1
+        with:
+          license: 'Personal'
+          username: ${{ secrets.UNITY_USERNAME }}
+          password: ${{ secrets.UNITY_PASSWORD }}
+          # required for Professional license activations
+          # serial: ${{ secrets.UNITY_SERIAL }}
+      - uses: buildalon/unity-action@v1
+        name: Project Validation
+        with:
+          log-name: 'project-validation'
+          args: '-quit -batchmode -executeMethod Utilities.Editor.BuildPipeline.UnityPlayerBuildTools.ValidateProject'
+      - uses: buildalon/unity-action@v1
+        name: '${{ matrix.build-target }}-Build'
+        with:
+          log-name: '${{ matrix.build-target }}-Build'
+          build-target: '${{ matrix.build-target }}'
+          args: '-quit -batchmode -executeMethod Utilities.Editor.BuildPipeline.UnityPlayerBuildTools.StartCommandLineBuild${{ matrix.build-args }}'
+      - uses: buildalon/unity-xcode-builder@v1
+        if: matrix.os == 'macos-13' && success()
```

Comment:
check the build-target instead so we don't have to replace macos-13 here for buildalon